osu.Game/Screens/Play/PauseContainer.cs

```diff
@@ -126,14 +126,52 @@ public class PauseOverlay : MenuOverlay
             public override string Header => "paused";
             public override string Description => "you're not going to do what i think you're going to do, are ya?";
 
+            private int selectedButton;
+
             protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
             {
                 if (!args.Repeat && args.Key == Key.Escape)
                 {
                     Buttons.Children.First().TriggerOnClick();
                     return true;
                 }
-
+                if (!args.Repeat && args.Key == Key.Up)
+                {
+                    if(selectedButton > 0)
+                    {
+                        Buttons[selectedButton].IsSelected = false;
+                        selectedButton--;
+                        Buttons[selectedButton].IsSelected = true;
+                    }
+                    return true;
+                }
+                if (!args.Repeat && args.Key == Key.Down)
+                {
+                    if (selectedButton < Buttons.Count - 1)
+                    {
+                        Buttons[selectedButton].IsSelected = false;
+                        selectedButton++;
+                        Buttons[selectedButton].IsSelected = true;
+                    }
+                    return true;
+                }
+                if (!args.Repeat && args.Key == Key.Enter)

```

Comment:
How about instead of handling this `Key.Enter` branch both here and in FailOverlay, having each dialog button handle `Key.Enter` if `IsSelected = true`?